sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -946,7 +946,11 @@ object SparkSession extends Logging {
           SparkContext.getOrCreate(sparkConf)
           // Do not update `SparkConf` for existing `SparkContext`, as it's shared by all sessions.
         }
-
+        // We should reset `spark.sql.catalogImplementation` according to current requirement.
+        if (sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "in-memory" &&
+          sparkConf.get(CATALOG_IMPLEMENTATION) == "hive") {
```

Comment:
What if `sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "hive" && sparkConf.get(CATALOG_IMPLEMENTATION) == "in-memory"`?

Reply:
> What if `sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "hive" && sparkConf.get(CATALOG_IMPLEMENTATION) == "in-memory"`?

I think if `sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "hive"`, I think user have set this before start SparkSession.

And we have a method `HiveUtils. withHiveExternalCatalog` we should respect this too ==

Reply:
> I think if sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "hive", I think user have set this before start SparkSession.

`CATALOG_IMPLEMENTATION` is a static configuration, so we can change the value that a user sets before starting a SparkContext?

Reply:
> > I think if sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "hive", I think user have set this before start SparkSession.
> 
> `CATALOG_IMPLEMENTATION` is a static configuration, so we can change the value that a user sets before starting a SparkContext?

But if we start SparkContext and then call `HiveUtils. withHiveExternalCatalog()` then use this SparkContext to start SparkSession?

Seems our test framework is build like this. and you can see my test result before the last finished one.